### PR TITLE
WIP: compute: use timely's container builder

### PR DESCRIPTION
**Pull request by nrainer-materialize**

### CODE REVIEW

File: src/compute/src/render/join/mz_join_core.rs [hunk 1]
```diff
@@ -576,31 +575,27 @@ where
...
         yield_fn: YFn,
         work: &mut usize,
     ) where
-        I: IntoIterator<Item = D>,
-        L: FnMut(C1::Key<'_>, C1::Val<'_>, C2::Val<'_>) -> I,
+        L: FnMut(C1::Key<'_>, C1::Val<'_>, C2::Val<'_>, &mut JoinSession<T, CB, CB::Container>),
         YFn: Fn(usize) -> bool,
-        C: PushContainer,
-        (D, T, Diff): PushInto<C>,
+        <CB as ContainerBuilder>::Container: ContainerBuilder,
```
Comment (antiguru) [inline src/compute/src/render/join/mz_join_core.rs:587]: This constraint looks off, a container should not be a builder. Probably caused by the `EffortBuilder<CB::Container>` type further up.

**Review by antiguru — COMMENTED** [review]
Thanks, looks like it's heading in the right direction!

**Comment by nrainer-materialize** [comment]
> Thanks, looks like it's heading in the right direction!

Thanks for the hints!

**Comment by nrainer-materialize** [comment]
@antiguru: Will I need to adjust `src/operators/arrange/arrangement.rs:241` in differential-dataflow to be able to complete this task?

**Comment by antiguru** [comment]
I haven't checked, but you should not.

**Comment by nrainer-materialize** [comment]
Notes to self:
* https://github.com/TimelyDataflow/differential-dataflow/pull/477/files
* https://github.com/TimelyDataflow/differential-dataflow/pull/478/files#diff-39970781c47508b22a534d807d01f7a6d3825f396b6b4bf5d1db8ad50f516906R370

**Comment by nrainer-materialize** [comment]
Closing stale PR.

**Pull request closed without merging**